File: Lecture-7/Lecture/Code/DynamicFieldImpl1_Part2.java
  ...

  public void append(T item) {
    T[] @{\color{java_variable}newElements}@ = new T[size + 1];
    
    // Copy elements to new array
    for (int i = 0; i < size; i++) {
      @{\color{java_variable}newElements}@[i] = elements[i];
    }
    
    elements = @{\color{java_variable}newElements}@;
    size++;
  }
}

File: Lecture-7/Lecture/Code/DynamicFieldImpl1_Part2.cpp
void DynamicArray::append(T item) {
  T[] newElements = new T[size + 1];

  // Copy elements to new array
  for (int i = 0; i < size; i++) {
    newElements[i] = elements[i];
  }

  delete[] elements;
  elements = newElements;
  size++;
}